elasticapm/traces.py

```diff
@@ -709,13 +744,52 @@ def set_transaction_name(name, override=True):
 
 
 def set_transaction_result(result, override=True):
+    """
+    Sets the result of the transaction. The result could be e.g. the HTTP status class (e.g "HTTP 5xx") for
+    HTTP requests, or "success"/"fail" for background tasks.
+
+    :param name: the name of the transaction
+    :param override: if set to False, the name is only set if no name has been set before
+    :return: None
+    """
+
     transaction = execution_context.get_transaction()
     if not transaction:
         return
     if transaction.result is None or override:
         transaction.result = result
 
 
+def set_transaction_success():
+    """
+    Marks this transaction as successful. This should only be done at the end of a transaction
+    when the outcome is determined.
+
+    This value is used for error rate calculations.
+
+    :return: None
+    """
+    transaction = execution_context.get_transaction()
+    if not transaction:
+        return
+    transaction.set_success()
```

Comment:
It looks like this the outcome may get overridden by the agent after on transaction end.

The spec states
> This value must always take precedence over the automatically determined value.

Reply:
You're right, I remembered something wrong. I thought we check first if `transaction.outcome != "unknown"`, but that happens at a different place. I'll add that check to the `set_success` and `set_failure` setters.